dotnet/ToDoApp/ToDoApp/ViewModel/WelcomeViewModel.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Realms;
+using Realms.Sync;
+using Xamarin.Forms;
+
+namespace ToDoApp
+{
+    // View model responsible for user sign up and sign in.
+    public class WelcomeViewModel : INotifyPropertyChanged
+    {
+        private string _message;
+        private string _username;
+        private string _password;
+        private bool _isFormEnabled = true;
+        private INavigation _navigation;
+
+        public string Username
+        {
+            get { return _username; }
+            set
+            {
+                if (_username == value)
+                {
+                    return;
+                }
+                _username = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public string Password
+        {
+            get { return _password; }
+            set
+            {
+                if (_password == value)
+                {
+                    return;
+                }
+                _password = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public string Message
```

Comment:
I know it's not as user friendly, but for the sake of simplicity, you can just alert the error rather than displaying a message. Actually, you can probably replace this entire viewmodel with `UserDialogs.Instance.LoginAsync`. Again, it's obviously not best practice, but for the sake of focusing mostly on Realm API, I would strip anything not Realm related as much as possible.